tools/tm-bench/main.go

```diff
@@ -93,17 +93,31 @@ Examples:
 	)
 	logger.Info("Latest block height", "h", initialHeight)
 
-	// record time start
-	timeStart := time.Now()
-	logger.Info("Time started", "t", timeStart)
-
 	transacters := startTransacters(
 		endpoints,
 		connections,
 		txsRate,
 		txSize,
 		"broadcast_tx_"+broadcastTxMethod,
 	)
+
+	// Wait until transacters have begun until we get the start time
+	for {
+		started := true
+		for _, t := range transacters {
+			for i := 0; i < t.Connections; i++ {
+				if !t.connsStarted[i] && !t.connsBroken[i] {
```

Comment:
it's not good that we're accessing private fields here. We should move this logic back to transacter: either by waiting in the end of `.OnStart` or by creating a separate method

```
for _, t := range transacters {
  t.WaitUntilAllConnectionsStartFiringTxs()
}
```

I slightly prefer the first option since you don't have to call a separate method. This is given we add a comment to OnStart saying OnStart blocks until all connections are up and firing txs.

Reply:
Note: you can use `sync.WaitGroup` instead of for loop to wait until all connections are sending txs.

Reply:
I thought about using the first method with a sync.WaitGroup, but I think its desirable to want the transacters to be able to start up in parallel in some capacity, so I'll add the method to the transacter which you suggested!

Reply:
I don't follow. How waiting in OnStart until we're sending txs permits anybody from spinning as many transacters as he/she wants.

you can do

```
        wg.Add(len(endpoints))
	for i, e := range endpoints {
		t := newTransacter(e, connections, txsRate, txSize, broadcastTxMethod)
                go func() {
                  defer wg.Done()
                   t.Start()
                }
```

Reply:
Oh I see. I was using the wait group in the Start method of each transacter method. You're right, I'll change the code to use that.